framework/sessions/__init__.py

```diff
@@ -45,6 +48,27 @@ def add_key_to_url(url, scheme, key):
     parsed_redirect_url = parsed_url._replace(**replacements)
     return urlparse.urlunparse(parsed_redirect_url)
 
+def get_node_id(url):
```

Comment:
I think this will break if a user tries to view a file with a VOL. 

Files can have their own guid, but `if file.node in private_link.nodes`, the link should still be granted view permission.